tests/test_units/test_table.py

```diff
@@ -8,25 +8,49 @@
 from fmu.dataio._objectdata_provider import _ObjectDataProvider
 
 
-def assert_works(file_path, answer):
-    """does the assert work for all tests
-
+def read_dict(file_path):
+    """Reads text file into dictionary
     Args:
         file_path (string): path to generated file
-        answer (list): expected answer
+    Returns:
+        dict: contents of file
     """
     file_path = Path(file_path)
     meta_path = file_path.parent / f".{file_path.name}.yml"
     meta = yaml_load(meta_path)
     file_path.unlink()
     meta_path.unlink()
-    index = meta["data"]["table_index"]
+    return meta
+
+
+def do_the_assert(index, answer, field_to_check):
+    """Assert that index is list and the answer is correct
+
+    Args:
+        index (should be list): what to check
+    """
     type_failure = f"{index} should be list, but is {type(index)}"
-    fail_string = f"table index should be {answer}, but is {index}"
+    fail_string = f"{field_to_check} should be {answer}, but is {index}"
     assert isinstance(index, list), type_failure
     assert index == answer, fail_string
 
 
+def assert_works(dict_input, answer, field_to_check="table_index"):
```

Comment:
Have you considered the `@pytest.mark.parametrize` pattern here instead?

Reply:
No, but have rewritten some of the tests, so no defaulted values now. If that is what you were referring to.

Reply:
I was thinking to use `parametrize` instead of having separate functions for calling the actual test. See example here: https://github.com/equinor/fmu-dataio/blob/d7a937ba852266b881140c615ab09bd4206fc31f/tests/test_units/test_dataio.py#L91

E.g. instead of explicitly calling another function for each input, `pytest` does it for you when you decorate with `@pytest.mark.parametrize`